scripts/yum-plugins/ptoken.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/python
+
+# Drop this file into /usr/lib/yum-plugins/
+# Enable it by creating conf file /etc/yum/pluginconf.d/ptoken.conf:
+#   [main]
+#   enabled=1
+#
+# Configure it by:
+# yum-config-manager --setopt=<repo-name>.ptoken=file://<token-file-path> --save
+
+from yum import config
+from yum.plugins import TYPE_CORE
+
+requires_api_version = '2.5'
+plugin_type = (TYPE_CORE,)
+
+def config_hook(conduit):
+    config.RepoConf.ptoken = config.BoolOption(False)
+
+def init_hook(conduit):
+    with open('/etc/xensource/ptoken') as f:
+        ptoken = f.read().strip()
+
+    repos = conduit.getRepos()
+    for name in repos.repos:
+        repo = repos.repos[name]
```

Comment:
Is it worth to restrict the repo URL to "http://127.0.0.1:8000"?

Reply:
How exactly? Something like `if repo.getConfigOption('baseurl') != "http://127.0.0.1:8000": continue`?

Reply:
Do we have to worry about the token containing anything that would need escaping when passed in an URL? Or would whoever constructs the URL take care of this?

Reply:
It goes into a cookie header, not the URL. And it's a fixed format, defined by xapi, that doesn't have weird characters.

Reply:
> How exactly? Something like `if repo.getConfigOption('baseurl') != "http://127.0.0.1:8000": continue`?

Yes. Maybe we have to have some hard-coding in this plugin.

Reply:
I've now added a restriction for the baseurl to localhost.